### ISSUE TITLE
Confirm link in e-mail should redirect to signed-up site (bugzilla 927343)

### ISSUE THREAD

**Issue by jaredhirsch** [mozilla/persona#3983]
From "rubbish@gmx.net":

User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:25.0) Gecko/20100101 Firefox/25.0 (Beta/Release)
Build ID: 20131010180222

Steps to reproduce:

Sign-up for the first time with a not identity-bridged e-mail address. Persona sends an e-mail with a link to confirm the e-mail address.

Actual results:

Clicking on the confirm link opens the Persona website.

Expected results:

Clicking on the confirm link should open the website, the user signed-up to.

We get many requests of users who are confused to see the Persona website instead of ours. They wonder what this website is and why it is shown and how to get back to our website.

When signing up to our website, the user expects to get back to where he started after the whole sign-up process.

A solution would be to add an option "successRedirect" (or something like that) to the id.get() options map. If set, Persona should redirect to that URL after confirming the e-mail address.

**Comment by jaredhirsch** [comment]
I've ported the bug but don't have the github creds for the user who posted

**Comment by jaredhirsch** [comment]
OK, answering the question: if you use navigator.id.watch instead of navigator.id.get, you can pass a `returnTo` parameter in your navigator.id.request call, and get users redirected back. This is one of the advantages of using `watch`.

https://developer.mozilla.org/en-US/docs/Web/API/navigator.id.request

https://developer.mozilla.org/en-US/docs/Web/API/navigator.id.watch

Closing.

**Comment by jkissel** [comment]
Is there a technical reason for the Callback API not having this option? If not I would appreciate consistency.

But for now, the Observer API is OK for my case. Thank you :-) I have to refactor my whole login process, though ...

**Comment by shane-tomlinson** [comment]
@jkissel - yup, the technical reason is that .get/.getVerifiedEmail MUST be called from within a click/keypress handler to avoid the browser's popup blocker. Since we want users to be automatically signed in after verifying their email address, we had to come up with a solution that allowed sites to express their desire to have users be logged in automatically without requiring the user to take any action. This is why we came up with the navigator.id.request/navigator.id.watch API.
